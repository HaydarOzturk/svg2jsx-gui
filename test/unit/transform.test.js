const transform = require('../../packages/transform');

describe('transform tests', () => {
  test('clean attributes', async () => {
    const transformed = await transform('<svg id=" test " strok="blue" fill="   red" />');

    expect(transformed).toEqual(expect.not.stringMatching(/strok/));
    expect(transformed).toMatchSnapshot();
  });

  test('remove unnecessary nodes', async () => {
    const transformed = await transform('<svg id="test"><g><g></g></g></svg>');

    expect(transformed).toEqual(expect.stringMatching(/<svg \/>/));
    expect(transformed).toMatchSnapshot();
  });

  test('convert attributes', async () => {
    const transformed = await transform('<svg class="test" data-name="svg" fill-rule="evenodd"/>');

    expect(transformed).toEqual(expect.stringMatching(/className=/));
    expect(transformed).toEqual(expect.not.stringMatching(/class=/));
    expect(transformed).toEqual(expect.stringMatching(/fillRule=/));
    expect(transformed).toEqual(expect.not.stringMatching(/fill-rule=/));
    expect(transformed).toEqual(expect.stringMatching(/data-name=/));
    expect(transformed).toEqual(expect.not.stringMatching(/dataName=/));
    expect(transformed).toMatchSnapshot();
  });

  test('convert inline styles to style objects', async () => {
    const transformed = await transform('<svg style="margin-left:20px;padding-top:20px;" />');

    expect(transformed).toEqual(expect.stringMatching(/marginLeft: 20/));
    expect(transformed).toEqual(expect.stringMatching(/paddingTop: 20/));
    expect(transformed).toMatchSnapshot();
  });
});
