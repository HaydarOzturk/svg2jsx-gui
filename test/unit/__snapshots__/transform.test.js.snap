// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform tests clean attributes 1`] = `
"<svg fill=\\"red\\" />;
"
`;

exports[`transform tests convert attributes 1`] = `
"<svg className=\\"test\\" data-name=\\"svg\\" fillRule=\\"evenodd\\" />;
"
`;

exports[`transform tests convert inline styles to style objects 1`] = `
"<svg style={{ marginLeft: 20, paddingTop: 20 }} />;
"
`;

exports[`transform tests remove unnecessary nodes 1`] = `
"<svg />;
"
`;
